hummingbot/connector/exchange/bittrex/bittrex_exchange.py

```diff
@@ -0,0 +1,403 @@
+import asyncio
+import datetime
+from decimal import Decimal
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Tuple
+
+from bidict import bidict
+
+from hummingbot.connector.constants import s_decimal_NaN
+from hummingbot.connector.exchange.bittrex import (
+    bittrex_constants as CONSTANTS,
+    bittrex_utils,
+    bittrex_web_utils as web_utils,
+)
+from hummingbot.connector.exchange.bittrex.bittrex_api_order_book_data_source import BittrexAPIOrderBookDataSource
+from hummingbot.connector.exchange.bittrex.bittrex_api_user_stream_data_source import BittrexAPIUserStreamDataSource
+from hummingbot.connector.exchange.bittrex.bittrex_auth import BittrexAuth
+from hummingbot.connector.exchange_py_base import ExchangePyBase
+from hummingbot.connector.trading_rule import TradingRule
+from hummingbot.connector.utils import combine_to_hb_trading_pair, split_hb_trading_pair
+from hummingbot.core.data_type.common import OrderType, TradeType
+from hummingbot.core.data_type.in_flight_order import InFlightOrder, OrderUpdate, TradeUpdate
+from hummingbot.core.data_type.order_book_tracker_data_source import OrderBookTrackerDataSource
+from hummingbot.core.data_type.trade_fee import TokenAmount, TradeFeeBase
+from hummingbot.core.data_type.user_stream_tracker_data_source import UserStreamTrackerDataSource
+from hummingbot.core.utils.async_utils import safe_ensure_future
+from hummingbot.core.utils.estimate_fee import build_trade_fee
+from hummingbot.core.web_assistant.web_assistants_factory import WebAssistantsFactory
+
+if TYPE_CHECKING:
+    from hummingbot.client.config.config_helpers import ClientConfigAdapter
+
+
+class BittrexExchange(ExchangePyBase):
+
+    web_utils = web_utils
+
+    def __init__(self,
+                 client_config_map: "ClientConfigAdapter",
+                 bittrex_api_key: str,
+                 bittrex_secret_key: str,
+                 trading_pairs: Optional[List[str]] = None,
+                 trading_required: bool = True):
+        self.api_key = bittrex_api_key
+        self.secret_key = bittrex_secret_key
+        self._trading_required = trading_required
+        self._trading_pairs = trading_pairs
+        super().__init__(client_config_map)
+
+    @property
+    def name(self) -> str:
+        return "bittrex"
+
+    @property
+    def authenticator(self) -> BittrexAuth:
+        return BittrexAuth(api_key=self.api_key,
+                           secret_key=self. secret_key,
+                           time_provider=self._time_synchronizer)
+
+    @property
+    def rate_limits_rules(self):
+        return CONSTANTS.RATE_LIMITS
+
+    @property
+    def domain(self):
+        return CONSTANTS.DEFAULT_DOMAIN
```

Comment:
Lets just return an empty string here instead. Doing this seems to suggest there is in fact a domain we are using for Bittrex

Reply:
As suggested by Abel, the gold standard is to have the default domain defined in constants even if it is empty string